### PR TITLE
More GUI Implementation

### PR DESCRIPTION
**Pull request by AdamG-L**
- GUI ADD/EDIT/DELETE Field should all be working now
- Added custom edit method to fields for the GUI
- Delete functionality can be reused
- Connecting specific LineEdits within a connect call can be reused

### CODE REVIEW

File: src/umleditor/mvc_model/entity.py [hunk 1]
```diff
@@ -48,6 +48,22 @@ def add_field(self, field_name:str):
             raise CustomExceptions.FieldExistsError(field_name)
         else:
             self._fields.append(field_name)
+    
+    def edit_field(self, old_field: str, new_field: str):
```
Comment (pwfreedm) [inline src/umleditor/mvc_model/entity.py:52]: why does the method rename_field not work here?

Comment (pwfreedm) [reply src/umleditor/mvc_model/entity.py:52]: ^if the method rename_field works here, delete this method.

Comment (AdamG-L) [reply src/umleditor/mvc_model/entity.py:52]: The edit_field allows for passing an old and new name without errors, while the rename does not, this functionality is necessary with the current GUI Design.

Comment (pwfreedm) [reply src/umleditor/mvc_model/entity.py:52]: To be clear, we are both talking about the rename_field on line 71 of entity.py, correct? If so, the functions are literally identical. Error checking doesn't matter if the error cannot be raised, so there's no reason to make a separate method to exclude it.

Comment (AdamG-L) [reply src/umleditor/mvc_model/entity.py:52]: Yes, I am referring to the  def rename_field() method. The key difference between edit and rename is that within rename it is considered illegal to rename a field given the same name, hence it raises an error. My edit_field function does not consider this action illegal and will not throw an exception. I.E if I want to edit something and I don't wind up changing the values, that's okay within edit_field. It is not okay to perform this edit within rename_field() and will stop the flow of the GUI. Although the difference in code is minor, the methods are not interchangeable within the GUI.

Comment (pwfreedm) [reply src/umleditor/mvc_model/entity.py:52]: you can add "if old_name == new_name: return" to rename_field. Setters conventionally do nothing if old and new are the same. It won't break the CLI.

Comment (AdamG-L) [reply src/umleditor/mvc_model/entity.py]: rename_field has been modified and edit_field and related files have been deleted

**Review by pwfreedm — APPROVED** [review]

**Pull request merged**